File: pyjl/tests_py2many/with_open.py
#!/usr/bin/env python3

from tempfile import NamedTemporaryFile


if __name__ == "__main__":
    with NamedTemporaryFile(mode="a+", delete=False) as temp_file:
        file_path = temp_file.name
        with open(file_path, "w") as f:
            f.write("hello")
        with open(file_path, "r") as f:
            assert f.read(1) == "h"
            assert f.read() == "ello"
            print("OK")

File: tests/expected/julia/with_open.jl
using tempfile: NamedTemporaryFile
function main()
    NamedTempFile::new() do temp_file
        file_path = name(temp_file)
        open(file_path, "w") do f
            write_(f, "hello")
        end
        open(file_path, "r") do f
            @assert(read(f, 1) == "h")
            @assert(read(f) == "ello")
            println("OK")
        end
    end
end

main()
